File: app/src/main/java/com/example/myapplication/MySiteApi.java
package com.example.myapplication;
import java.util.ArrayList;
import java.util.HashMap;

import Models.ViewModel.MainPageData;
import Models.linked.Product;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.PATCH;
import retrofit2.http.POST;
import retrofit2.http.Query;
public interface MySiteApi {
    @GET("api/newlyAdded")
    public Call<MainPageData> getNewlyAdded();

    @GET("api/catalog")
     Call<ArrayList<Product>> GetCatalog(@Query("type")String type);

    @GET("api/GetProduct")
    Call<Product> GetProduct(@Query("id") int id);

    @POST("api/Login")
    Call<String> SignIn(@Body HashMap<String, String> SignData);

    @GET("api/ShoppingCart")
    Call<ArrayList<Product>> GetShoppingCart();

    @POST("api/ShoppingCart")
    Call<String> PostSHoppingCart(@Body HashMap<String, Integer> data);

    @DELETE("api/ShoppingCart")
    Call<String> DeleteShoppingCart(@Query("id") int id);

    @PATCH("api/CompleteOrder")
    Call<String> CompleteOrder();
}

File: app/src/main/java/com/example/myapplication/MySiteApi.kt
package com.example.myapplication

import Models.linked.Product
import retrofit2.Call
import retrofit2.Response
import retrofit2.http.*
import java.util.ArrayList
import java.util.HashMap

interface MySiteApi {
    @GET("api/newlyAdded")
    suspend fun newlyAdded(@Query("skip") skip: Int, @Query("Count") count: Int): Response<ArrayList<Product>?>?

    @GET("api/MostBuyed")
    suspend fun MostBuyed(@Query("skip") skip: Int, @Query("Count") count: Int): Response<ArrayList<Product>?>?

    @GET("api/MostDiscounted")
    suspend fun MaxDiscounted(@Query("skip") skip: Int, @Query("Count") count: Int): Response<ArrayList<Product>?>?

    @GET("api/catalog")
    suspend fun GetCatalog(@Query("type") type: String?, @Query("skip") skip: Int, @Query("Count")count: Int): Response<ArrayList<Product>?>?

    @GET("api/GetProduct")
    suspend fun GetProduct(@Query("id") id: Int): Response<Product>?

    @POST("api/accountApi/Login")
    suspend fun SignIn(@Body SignData: HashMap<String, String>?): Response<String>?

    @POST("api/accountApi/Register")
    suspend fun Register(@Body RegData : HashMap<String, String>?): Response<String>?

    @GET("api/ShoppingCart")
    fun GetShoppingCart(): Call<ArrayList<Product>?>?

    @POST("api/ShoppingCart")
    fun PostSHoppingCart(@Body data: HashMap<String, Int>?): Call<String>?

    @DELETE("api/ShoppingCart")
    fun DeleteShoppingCart(@Query("id") id: Int): Call<String>?

    @PATCH("api/CompleteOrder")
    fun CompleteOrder(): Call<String>?
}